beetsplug/subsonicupdate.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+# Copyright 2016, Adrian Sampson.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Updates Subsonic library on Beets import
+Your Beets configuration file should contain
+a "subsonic" section like the following:
+    subsonic:
+        host: 192.168.x.y (Subsonic server IP)
+        port: 4040 (default)
+        user: <your username>
+        pass: <your password>
+"""
+from __future__ import division, absolute_import, print_function
+
+from beets.plugins import BeetsPlugin
+import requests
+import string
+import hashlib
+import random
+
+__author__ = 'https://github.com/maffo999'
+
+
+class SubsonicUpdate(BeetsPlugin):
+    def __init__(self):
+        super(SubsonicUpdate, self).__init__()
+
+# Set default configuration values
+        self.config['subsonic'].add({
```

Comment:
I'm not sure you mean `self.config['subsonic']` here, which will expect something like this in the configuration YAML:

```yaml
subsonicupdate:
    subsonic:
        host: ...
        port: ...
```

You can either use `self.config` alone (to put the options directly under the plugin name) or use the global configuration, as in `config['subsonic']` after importing the global `config` object, to create a top-level `subsonic` entry.

Reply:
This was due to my misunderstanding of how the config works, removed the "self" references and imported "config" from beets.